src/IECoreDelight/Renderer.cpp

```diff
@@ -348,6 +373,19 @@ class DelightOutput : public IECore::RefCounted
 				m_layerHandle.name(), "outputdrivers",
 				0, nullptr
 			);
+
+			if( lightGroup != "" )
+			{
+				for( string lightGroupToken : lightGroupTokens )
+				{
+					NSIConnect(
+						m_context,
+						lightGroupToken.c_str(), "",
+						m_layerHandle.name(), "lightset",
+						0, nullptr
+					);
+				}
+			}
```

Comment:
I fear that this won't scale to real production scenarios - there will just be too many lights for it to be practical to manage them as a list of individual locations. It's also really vulnerable to lights changing names, being pruned etc.

I think a production-proof solution would either have to use Gaffer sets to define light groups, or a sort of "virtual parameter" on the light to say what groups it was in (mimicking the Arnold approach).

Reply:
Completely agreed. Similar to `customdrivername` this was a first pass setup to get light groups running with 3Delight and it definitely isn't an elegant solution. 

I did want to see what it would take to get Gaffer sets translated to NSI sets, since that would not only help light groups, but any future attempt at light linking implementation as well. Once we have sets working well, I'll revisit light groups and any other potentially needed changes like the above mentioned `GafferScene/Render.cpp` outputs order.

Reply:
It's not very well documented, but any sets with names matching `render:*` are exported to renderers as a string attribute with the list of sets an object belongs to. It would be possible to map that to connections to set nodes in the NSI backend. Maybe that's a starting point?

Reply:
That's great to know. Thanks for the info!